### PR TITLE
Image compressor play created (#721)

### LINKED ISSUE #721: [Add a Play]: Image Compressor
### What's the Play about?

App, where user will upload large image size and it will compress it down to small size.

### What ReactJS concept will be used to create this play?

JSX, UseState, UseEffect
For compression I will going to use `browser-image-compression` npm package.

### Are you willing to take it up for implementation?

Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by aks318**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description
Image compressor play added #721 

Fixes #721 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/plays/image-compressor/ImageCompressor.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      return;
+    }
+    setSelectedFile(imageFile);
+    setSelectedFileSize((imageFile.size / 1024 / 1024).toFixed(2));
+    setCompressing(true);
+    const options = {
+      maxSizeMB: 1,
+      maxWidthOrHeight: 1920,
+      useWebWorker: true,
+    };
```
Comment (Sachin-chaurasiya) [inline src/plays/image-compressor/ImageCompressor.js:28]: Move this to constant.

File: src/plays/image-compressor/ImageCompressor.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import imageCompression from "browser-image-compression";
+import CircularProgress from "@mui/material/CircularProgress";
+import "./styles.css";
+
+function ImageCompressor(props) {
+  // Your Code Start below.
+
+  const [selectedFile, setSelectedFile] = useState(null);
+  const [selectedFileSize, setSelectedFileSize] = useState(null);
+  const [compressing, setCompressing] = useState(false);
```
Comment (Sachin-chaurasiya) [inline src/plays/image-compressor/ImageCompressor.js:12]:
```suggestion
  const [isCompressing, setCompressing] = useState(false);
```

File: src/plays/image-compressor/ImageCompressor.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  const [compressedFile, setCompressedFile] = useState(null);
+  const [compressedFileSize, setCompressedFileSize] = useState(null);
+
+  const handleImageUpload = async (event) => {
+    const imageFile = event.target.files[0];
+    if (!imageFile) {
+      return;
+    }
+    setSelectedFile(imageFile);
+    setSelectedFileSize((imageFile.size / 1024 / 1024).toFixed(2));
```
Comment (Sachin-chaurasiya) [inline src/plays/image-compressor/ImageCompressor.js:22]: Move this logic to util and use it everywhere.
```js
(imageFile.size / 1024 / 1024).toFixed(2)
````

File: src/plays/image-compressor/ImageCompressor.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      maxWidthOrHeight: 1920,
+      useWebWorker: true,
+    };
+    try {
+      const compressedFile = await imageCompression(imageFile, options);
+      setCompressedFile(compressedFile);
+      setCompressedFileSize((compressedFile.size / 1024 / 1024).toFixed(2));
+      setCompressing(false);
+    } catch (error) {
+      console.log(error);
```
Comment (Sachin-chaurasiya) [inline src/plays/image-compressor/ImageCompressor.js:35]: Can we avoid adding a console log and show some kind of error message to the user?

File: src/plays/image-compressor/ImageCompressor.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+              type="file"
+              accept="image/*"
+              onChange={handleImageUpload}
+            />
+            <div className="imageCompressor__imageContainer">
+              <div className="imageCompressor__box">
+                <div className="imageCompressor__image">
+                  {selectedFile && (
+                    <img src={URL.createObjectURL(selectedFile)} alt="" />
+                  )}
```
Comment (Sachin-chaurasiya) [inline src/plays/image-compressor/ImageCompressor.js:57]:
```suggestion
                    <img src={URL.createObjectURL(selectedFile)} alt="someAltValue" />
                  )}
```

File: src/plays/image-compressor/ImageCompressor.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                <p>{selectedFileSize} MB</p>
+              </div>
+              <div className="imageCompressor__progress">
+                {compressing && <CircularProgress />}
+              </div>
+              <div className="imageCompressor__box">
+                <div className="imageCompressor__image">
+                  {compressedFile && (
+                    <img src={URL.createObjectURL(compressedFile)} alt="" />
+                  )}
```
Comment (Sachin-chaurasiya) [inline src/plays/image-compressor/ImageCompressor.js:68]: Move this logic to util and use it here and in other places.
```js
URL.createObjectURL(compressedFile)
```

File: src/plays/image-compressor/ImageCompressor.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+                </div>
+                <p>{compressedFileSize} MB</p>
+              </div>
+            </div>
+            {compressedFile && (
```
Comment (Sachin-chaurasiya) [inline src/plays/image-compressor/ImageCompressor.js:73]: We have the same check above `compressedFile &&`, can we refactor the code and use this check one time and render both the compressed image and download button?

File: src/plays/image-compressor/readme.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+Update your implementation idea and details here
+
+## Consideration
+
+Update all considerations(if any)
+
+## Resources
+
+Update external resources(if any)
```
Comment (Sachin-chaurasiya) [inline src/plays/image-compressor/readme.md:27]: Can we update these details as well?

File: src/plays/image-compressor/styles.css [hunk 9]
```diff
@@ -0,0 +1,52 @@
+/* enter stlyes here */
+.imageCompressor {
```
Comment (Sachin-chaurasiya) [inline src/plays/image-compressor/styles.css:2]: Can we use the play name prefix with classes?

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]

File: src/plays/image-compressor/ImageCompressor.js (hunk 7 above)
Comment (aks318) [reply src/plays/image-compressor/ImageCompressor.js:73]: Both the checks are in different div. I think its impossible to render both using one check.

**Comment by aks318** [comment]
@Sachin-chaurasiya  Changes has been done, please review.

File: src/plays/image-compressor/ImageCompressor.js (hunk 7 above)
Comment (Sachin-chaurasiya) [reply src/plays/image-compressor/ImageCompressor.js:80]: @aks318 , Can't we use a fragment here and render both div with one condition?

Example:

```js
{condition && <>
<div1></div1>
<div2></div2>
</> }

```

Comment (aks318) [reply src/plays/image-compressor/ImageCompressor.js:80]: image

@Sachin-chaurasiya , I could have done it if they are siblings. But according to UI, 1st check renders image and 2nd renders button.

Comment (Sachin-chaurasiya) [reply src/plays/image-compressor/ImageCompressor.js:80]: @aks318 , Both depend on the same condition, right?
Can you try the suggested change?

**Comment by Sachin-chaurasiya** [comment]
@aks318 , Can you address the pending comments?

File: src/plays/image-compressor/ImageCompressor.js (hunk 7 above)
Comment (Sachin-chaurasiya) [reply src/plays/image-compressor/ImageCompressor.js:80]: @aks318 , no worries, I got the point about why we can't do that.

**Review by Sachin-chaurasiya — APPROVED** [review]

File: src/plays/image-compressor/ImageCompressor.js (hunk 7 above)
Comment (aks318) [reply src/plays/image-compressor/ImageCompressor.js:80]: @Sachin-chaurasiya Just opened my laptop and saw your comment. Great to see you got my point.

**Review by atapas — APPROVED** [review]

**Pull request merged**